doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -668,186 +644,119 @@ Induction
-   generalizing :n:`{+ @ident}`, and then performing ``inversion``.
-
-.. tacv:: inversion @ident as @or_and_intropattern in {+ @ident}
-
-   This allows naming the hypotheses introduced in the context by
-   :n:`inversion @ident in {+ @ident}`.
-
-.. tacv:: inversion_clear @ident in {+ @ident}
-
-   Let :n:`{+ @ident}` be identifiers in the local context. This tactic behaves
-   as generalizing :n:`{+ @ident}`, and then performing ``inversion_clear``.
-
-.. tacv:: inversion_clear @ident as @or_and_intropattern in {+ @ident}
-
-   This allows naming the hypotheses introduced in the context by
-   :n:`inversion_clear @ident in {+ @ident}`.
-
-.. tacv:: dependent inversion @ident
-   :name: dependent inversion
+.. _inversion-intropattern-ex:
 
-   That must be used when :n:`@ident` appears in the current goal. It acts like
-   ``inversion`` and then substitutes :n:`@ident` for the corresponding
-   :n:`@@term` in the goal.
+.. example:: :tacn:`inversion` with :n:`as @or_and_intropattern`
 
-.. tacv:: dependent inversion @ident as @or_and_intropattern
-
-   This allows naming the hypotheses introduced in the context by
-   :n:`dependent inversion @ident`.
-
-.. tacv:: dependent inversion_clear @ident
-
-   Like ``dependent inversion``, except that :n:`@ident` is cleared from the
-   local context.
-
-.. tacv:: dependent inversion_clear @ident as @or_and_intropattern
-
-   This allows naming the hypotheses introduced in the context by
-   :n:`dependent inversion_clear @ident`.
-
-.. tacv:: dependent inversion @ident with @term
-   :name: dependent inversion … with …
-
-   This variant allows you to specify the generalization of the goal. It is
-   useful when the system fails to generalize the goal automatically. If
-   :n:`@ident` has type :g:`(I t)` and :g:`I` has type :g:`forall (x:T), s`,
-   then :n:`@term` must be of type :g:`I:forall (x:T), I x -> s'` where
-   :g:`s'` is the type of the goal.
-
-.. tacv:: dependent inversion @ident as @or_and_intropattern with @term
-
-   This allows naming the hypotheses introduced in the context by
-   :n:`dependent inversion @ident with @term`.
-
-.. tacv:: dependent inversion_clear @ident with @term
-
-   Like :tacn:`dependent inversion … with …` with but clears :n:`@ident` from the
-   local context.
-
-.. tacv:: dependent inversion_clear @ident as @or_and_intropattern with @term
+   .. coqtop:: reset all
 
-   This allows naming the hypotheses introduced in the context by
-   :n:`dependent inversion_clear @ident with @term`.
+      Inductive contains0 : list nat -> Prop :=
+      | in_hd : forall l, contains0 (0 :: l)
+      | in_tl : forall l b, contains0 l -> contains0 (b :: l).
+      Goal forall l:list nat, contains0 (1 :: l) -> contains0 l.
+      intros l H; inversion H as [ | l' p Hl' [Heqp Heql'] ].
 
-.. tacv:: simple inversion @ident
-   :name: simple inversion
+   .. tacn:: inversion_clear {| @ident | @natural } {? as @or_and_intropattern } {? in {+ @ident } }
```

Comment:
There is an indentation problem which makes this and the following tactics appear inside the example above.